test/e2e/operator.go

```diff
@@ -272,3 +277,102 @@ var _ = Describe("ARO Operator - Conditions", func() {
 		Expect(err).NotTo(HaveOccurred())
 	})
 })
+
+var _ = Describe("ARO Operator - Azure NSG Reconciler", func() {
+	var location, resourceGroup, subnet, vnet string
+	var correctNSG *string
+	var dummynsg mgmtnetwork.SecurityGroup
+	ctx := context.Background()
+	type subnetDescriptor struct {
+		resourceGroup string
+		vnetName      string
+		subnetName    string
+		location      string
+	}
+	enableReconcileNSG := func() {
+		instance, err := clients.AROClusters.AroV1alpha1().Clusters().Get(ctx, arov1alpha1.SingletonClusterName, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		if !instance.Spec.Features.ReconcileNSGs {
+			instance.Spec.Features.ReconcileNSGs = true
+			_, err = clients.AROClusters.AroV1alpha1().Clusters().Update(ctx, instance, metav1.UpdateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+		}
+	}
```

Comment:
Hm. Why do we have it disabled by default? Is it intentional?

cc @Makdaam

Reply:
Also https://github.com/Azure/ARO-RP/pull/1663 changes the relevant controller and the flag.

Reply:
> Hm. Why do we have it disabled by default? Is it intentional?

Ah, right. I guess this is due to ArchitectureVersion not being respected.

Reply:
> Also #1663 changes the relevant controller and the flag.

I see feature being changed to `ReconcileSubnets`, I don't mind changing it in this test, but need to understand priority and when each of these plan on being merged so appropriate changes can be made.